File: PHP/1 INPUT DATA VERIFICATION/9 LDAP Injection/php/MOPAS_1_LDAP_inj.php
<?php
/**
 *  MOPAS LDAP Injection Example 1
 */
$name = $_GET['name'];

$ds = ldap_connect('localhost', 389);
echo "ldap_connect result: $ds\n";

if (ldap_set_option($ds, LDAP_OPT_PROTOCOL_VERSION, 3)) {
	echo "Using LDAPv3\n";
} else {
	echo "Failed to set protocol version to 3\n";
}

if (!$ds) {
	die('connection error');
}

$bind = ldap_bind($ds);
//echo "ldap_bind result: $r\n";
if (!$bind) {
	die('ldap_bind error');
}

$filter = "(givenName=$name)";
$search_result = ldap_search($ds, 'ou=Users,dc=example,dc=com', $filter);
//echo "search_result: $search_result\n";

//echo "search_result count " . ldap_count_entries($ds, $search_result) . "\n";

echo "elements:";
$info = ldap_get_entries($ds, $search_result);
//var_dump($info);
echo $info[0]["dn"];

ldap_close($ds);
?>

File: Java/1 INPUT DATA VERIFICATION/9 LDAP Injection/java/MOPAS_1_LDAP_inj.java
/**
 *  MOPAS LDAP Injection Example 1
 */
import java.io.IOException;
import java.io.PrintWriter;
import java.util.Hashtable;
import javax.naming.Context;
import javax.naming.NamingEnumeration;
import javax.naming.directory.Attributes;
import javax.naming.directory.SearchControls;
import javax.naming.directory.SearchResult;
import javax.naming.ldap.InitialLdapContext;
import javax.naming.ldap.LdapContext;
import javax.servlet.ServletException;
import javax.servlet.annotation.WebServlet;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

@WebServlet(urlPatterns = {"/MOPAS_1_LDAP_inj"})
public class MOPAS_1_LDAP_inj extends HttpServlet {

    @Override
    public void doGet(HttpServletRequest request, HttpServletResponse response)
            throws IOException, ServletException {

        PrintWriter responseWriter = response.getWriter();

        String name = request.getParameter("name");

        try {
            Hashtable env = new Hashtable();
            env.put(Context.INITIAL_CONTEXT_FACTORY,"com.sun.jndi.ldap.LdapCtxFactory");
            env.put(Context.PROVIDER_URL, "ldap://localhost:10389/");
            env.put(Context.SECURITY_AUTHENTICATION, "none");

            LdapContext ctx = new InitialLdapContext(env, null);
            ctx.setRequestControls(null);
            String filter = "(givenName=" + name + ")";
            NamingEnumeration namingEnum = ctx.search("ou=Users,dc=example,dc=com", filter, new SearchControls());

            while (namingEnum.hasMore ()) {
                SearchResult result = (SearchResult) namingEnum.next();
                Attributes attrs = result.getAttributes();
                responseWriter.println(attrs.get("cn"));
            }
            namingEnum.close();
        } catch (Exception e) {
        }
    }
}